webserver/src/webserver.cpp

```diff
@@ -92,35 +102,68 @@ int main(int argc, char* argv[])
 
-  cc::webserver::ServerContext ctx(compassRoot, vm);
+  ServerContext ctx(compassRoot, vm);
   requestHandler.pluginHandler.configure(ctx);
+  requestHandler.updateServedPathsFromPluginHandler();
 
   //--- Start mongoose server ---//
 
-  cc::webserver::ThreadedMongoose server(vm["jobs"].as<int>());
-  server.setOption("listening_port", std::to_string(vm["port"].as<int>()));
-  server.setOption("document_root", vm["webguiDir"].as<std::string>());
-
-  // Check if certificate.pem exists in the workspace - if so, start SSL.
-  auto certPath = fs::path(vm["workspace"].as<std::string>()).append("certificate.pem");
-  if (boost::filesystem::is_regular_file(certPath))
-  {
-    LOG(info) << "Starting HTTPS listener based on 'certificate.pem'.";
-    server.setOption("ssl_certificate", certPath.native());
-  }
+#if MG_ENABLE_SSL
+  // Check if certificates exist in the workspace - if so, start SSL.
+  auto certPath =
+    fs::path(vm["workspace"].as<std::string>()).append("server.pem").native();
+  auto keyPath =
+    fs::path(vm["workspace"].as<std::string>()).append("server.key").native();
+  if (fs::is_regular_file(certPath) && fs::is_regular_file(keyPath))
+    LOG(info) << "Starting HTTPS listener with certificate 'server.pem' and "
+                 "private key 'server.key'.";
   else
   {
-    LOG(warning) << "No 'certificate.pem' found in '--workspace', server "
-                    "running over conventional HTTP!";
+    LOG(warning)
+      << "No 'server.pem' or 'server.key' found in '--workspace', server "
+         "running over conventional HTTP!";
+    certPath = keyPath = "";
   }
-
-  LOG(info) << "Mongoose web server starting on port "
-            << server.getOption("listening_port");
+#else
+  LOG(warning) << "CodeCompass server built without SSL support. Running over "
+                  "conventional HTTP!";
+  std::string certPath, keyPath;
+#endif // MG_ENABLE_SSL
+
+  std::string listenPort = std::to_string(vm["port"].as<int>());
+  LOG(info) << "Mongoose web server starting on port " << listenPort;
+  MongooseServer server{
+    std::move(listenPort), WEBGUI_DIR, std::move(certPath), std::move(keyPath)};
+  server.setHTTPServiceURIPaths(&requestHandler.servedPaths);
+
+  //--- Create the worker threads that generate responses ---//
+
+  std::unique_ptr<JobQueueThreadPool<HTTPRequest>> threads =
+    make_thread_pool<HTTPRequest>(
+      vm["jobs"].as<int>(),
+      [&server, &requestHandler](HTTPRequest& req_) {
+        // The executor object of the worker thread is to handle the request
+        // itself by the main handler dispatching it to the specific handler
+        // implementation.
+        requestHandler(req_);
+        server.wakeUp();
+      },
+      true);
+
+  // The callback function of the server itself is to push a received and
+  // understood request to the worker threads' queue.
+  server.setHTTPServiceRequestHandler(
+    std::bind( // NOLINT: Need to pass function obj!
```

Comment:
Clang-Tidy 10.0 would whine that I should pass a lambda instead of a bind... but unfortunately, the other side is a `std::function` to which a lambda with a capture group would not compile.

Reply:
Actually, Clang-Tidy was right for the warning, just the automatic fix-it didn't compile by itself. Starting from C++14, you can actually give a lambda with a capture to an `std::function`, just need to do copy initialisation instead of truly capturing. Which is okay, given we uniquely own the thread-pool.